pkg/antctl/transform/networkpolicy/transform_test.go

```diff
@@ -0,0 +1,329 @@
+
+package networkpolicy
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	cpv1beta "antrea.io/antrea/pkg/apis/controlplane/v1beta2"
+)
+
+func TestListTransform(t *testing.T) {
+	var timeA = metav1.Now()
+	var timeB = metav1.Time{Time: metav1.Now().Add(1)}
+	var timeC = metav1.Time{Time: metav1.Now().Add(2)}
+	tests := []struct {
+		name         string
+		npList       *cpv1beta.NetworkPolicyList
+		opts         map[string]string
+		sortedNPList *cpv1beta.NetworkPolicyList
+	}{
+		{
+			name: "sort by name",
+			npList: &cpv1beta.NetworkPolicyList{
+				Items: []cpv1beta.NetworkPolicy{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "a",
+							UID:  "abc",
+						},
+						SourceRef: &cpv1beta.NetworkPolicyReference{
+							Name: "a",
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "c",
+							UID:  "abb",
+						},
+						SourceRef: &cpv1beta.NetworkPolicyReference{
+							Name: "c",
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "b",
+							UID:  "aaa",
+						},
+						SourceRef: &cpv1beta.NetworkPolicyReference{
+							Name: "b",
+						},
+					},
+				},
+			},
+			opts: map[string]string{
+				"sort-by": "name",
+			},
+			sortedNPList: &cpv1beta.NetworkPolicyList{
+				Items: []cpv1beta.NetworkPolicy{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "a",
+							UID:  "abc",
+						},
+						SourceRef: &cpv1beta.NetworkPolicyReference{
+							Name: "a",
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "b",
+							UID:  "aaa",
+						},
+						SourceRef: &cpv1beta.NetworkPolicyReference{
+							Name: "b",
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "c",
+							UID:  "abb",
+						},
+						SourceRef: &cpv1beta.NetworkPolicyReference{
+							Name: "c",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "sort by uid",
+			npList: &cpv1beta.NetworkPolicyList{
+				Items: []cpv1beta.NetworkPolicy{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "a",
+							UID:  "abc",
+						},
+						SourceRef: &cpv1beta.NetworkPolicyReference{
+							Name: "a",
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "c",
+							UID:  "abb",
+						},
+						SourceRef: &cpv1beta.NetworkPolicyReference{
+							Name: "c",
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "b",
+							UID:  "aaa",
+						},
+						SourceRef: &cpv1beta.NetworkPolicyReference{
+							Name: "b",
+						},
+					},
+				},
+			},
+			opts: map[string]string{
+				"sort-by": "uid",
+			},
+			sortedNPList: &cpv1beta.NetworkPolicyList{
+				Items: []cpv1beta.NetworkPolicy{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "b",
+							UID:  "aaa",
+						},
+						SourceRef: &cpv1beta.NetworkPolicyReference{
+							Name: "b",
+						},
```

Comment:
You can define a few variables for these NPs and re-order them as you need in the `Items` in both `npList` and `sortedNPList`.

Reply:
okay